tests/e2e.sh

```diff
@@ -20,17 +20,49 @@ set -o pipefail
 KUBERNETES_VERSION=v1.8.0
 KUBE_STATE_METRICS_LOG_DIR=./log
 KUBE_STATE_METRICS_IMAGE_NAME='quay.io/coreos/kube-state-metrics'
-KUBE_STATE_METRICS_IMAGE_NAME_PATTERN='quay.io\/coreos\/kube-state-metrics'
 PROMETHEUS_VERSION=2.0.0
+E2E_SETUP_MINIKUBE=${E2E_SETUP_MINIKUBE-yes}
+E2E_SETUP_KUBECTL=${E2E_SETUP_KUBECTL-yes}
+E2E_SETUP_PROMTOOL=${E2E_SETUP_PROMTOOL-yes}
+MINIKUBE_DRIVER=${MINIKUBE_DRIVER:-none}
+SUDO=${SUDO-sudo}
 
 mkdir -p $KUBE_STATE_METRICS_LOG_DIR
 
-# setup a Kubernetes cluster
-curl -sLo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
+function finish() {
+    echo "calling cleaup function"
+    # kill kubectl proxy in background
+    kill %1 || true
+    kubectl delete -f kubernetes/ || true
+    kubectl delete -f tests/manifests/ || true
+}
+
+function setup_minikube() {
+    # setup a Kubernetes cluster
```

Comment:
drop this comment as it is just downloading the minikube binary. :)

Reply:
Addressed in 6a44a4c.